lib/dynamodb-document-client.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const assert = require('assert');
+
+const AwsDriverError = require('./aws-driver-error');
+const Driver = require('./driver');
+
+class DynamoDBDocumentClientDriver extends Driver {
+  constructor(options) {
+    const clientOpts = {
+      apiVersion: '2012-08-10'
+    };
+    super('DynamoDB.DocumentClient', clientOpts, options);
+  }
+
+  async batchGet(context, params) {
+    const results = {};
+    let callParams = Object.assign({}, params);
+    let attempts = 0;
+
+    /* eslint-disable no-await-in-loop */
+    while (attempts < (this._conf.attempts || 10)) {
+      attempts += 1;
+
+      const { Responses, UnprocessedKeys } = await this._call(context, callParams, 'batchGet');
+
+      Object.entries(Responses).forEach(([tableName, responses]) => {
+        results[tableName] = (results[tableName] || []).concat(responses);
+      });
+
+      if (Object.keys(UnprocessedKeys).length > 0) {
+        callParams = Object.assign({}, callParams, {
+          RequestItems: UnprocessedKeys
+        });
+      } else {
+        return { Responses: results };
+      }
+    }
+    /* eslint-enable no-await-in-loop */
+
+    throw new AwsDriverError({
+      message: 'unable to fetch all requested items after too many retries'
+    });
+  }
+
+  get(context, params) {
+    return this._call(context, params, 'get');
+  }
+
+  put(context, params) {
+    return this._call(context, params, 'put');
+  }
+
+  query(context, params) {
+    return this._call(context, params, 'query');
+  }
+
+  scan(context, params) {
+    return this._call(context, params, 'scan');
+  }
+
+  update(context, params) {
+    return this._call(context, params, 'update');
+  }
+
+  _call(context, params, method) {
+    assert(params, 'missing params');
```

Comment:
these kind of assertions are only really valuable to you, the library author, which testing should guard against.  to someone using the library, if they see this, the library has a bug, and its confusing

Reply:
`method` check - maybe, but not `params`. To me, this is a safety net. This code will fail anyway, but as a user, I'd prefer to see `missing params` vs something like `Cannot read property 'Key' of undefined`

Reply:
that's fair!